node/src/reactor/validator/tests.rs

```diff
@@ -30,22 +34,34 @@ impl TestChain {
     ///
     /// Generates secret keys for `size` validators and creates a matching chainspec.
     fn new(rng: &mut TestRng, size: usize) -> Self {
-        // Create a secret key for each validator.
         let keys: Vec<SecretKey> = (0..size).map(|_| SecretKey::random(rng)).collect();
+        let stakes = keys
+            .iter()
+            .map(|secret_key| (PublicKey::from(secret_key), rng.gen_range(100, 999)))
+            .collect();
+        Self::new_with_keys(rng, keys, stakes)
+    }
 
+    /// Instantiates a new test chain configuration.
+    ///
+    /// Takes a vector of bonded keys with specified bond amounts.
+    fn new_with_keys(
```

Comment:
Maybe `new_with_genesis_accounts`?. `_with_keys` sounds as if it was trying to create a node with those keys as … validator keys, or node keys.

Reply:
Ah, but it's only test. Not that important then.